wrangler-proto/src/main/java/io/cdap/wrangler/proto/workspace/v2/Workspace.java

```diff
@@ -159,9 +175,23 @@ public Builder setInsights(JsonObject insights) {
       return this;
     }
 
+    public Builder setNullabilityMap (HashMap<String, UserDefinedAction> nullabilityMap) {
+      this.nullabilityMap = nullabilityMap;
+      return this;
+    }
+
     public Workspace build() {
       return new Workspace(workspaceName, workspaceId, directives, createdTimeMillis, updatedTimeMillis, sampleSpec,
-                           insights);
+                           insights, nullabilityMap);
     }
   }
+
+  /**
+   * UserDefinedAction enum.
+   */
+  public enum UserDefinedAction {
```

Comment:
should move this to its own class instead of being nested.

Reply:
Done